src/Core/Content/Test/Product/DataAbstractionLayer/ProductStreamUpdaterTest.php

```diff
@@ -100,34 +102,163 @@ public function testIndexingDoesNotBreakOnInvalidProductStreamFilters(): void
         $this->productStreamUpdater->updateProducts([$productId], Context::createDefaultContext());
     }
 
-    private function createProduct(string $productId): void
+    public function testConsiderInheritanceVariants(): void
     {
-        $this->productRepository->create(
+        $activeStreamId = Uuid::randomHex();
+        $inActiveStreamId = Uuid::randomHex();
+
+        $writtenEvent = $this->productStreamRepository->create([
             [
-                [
-                    'id' => $productId,
-                    'productNumber' => $productId,
-                    'stock' => 1,
-                    'name' => 'Test',
-                    'active' => true,
-                    'price' => [
-                        [
-                            'currencyId' => Defaults::CURRENCY,
-                            'gross' => 100,
-                            'net' => 9, 'linked' => false,
-                        ],
-                    ],
-                    'manufacturer' => ['name' => 'test'],
-                    'tax' => ['taxRate' => 19, 'name' => 'with id'],
-                    'visibilities' => [
-                        ['salesChannelId' => $this->salesChannel->getSalesChannel()->getId(), 'visibility' => ProductVisibilityDefinition::VISIBILITY_ALL],
-                    ],
-                    'categories' => [
-                        ['id' => Uuid::randomHex(), 'name' => 'Clothing'],
-                    ],
+                'id' => $activeStreamId,
+                'name' => 'test-inheritance',
+                'filters' => [
+                    [
+                        'type' => 'equals',
+                        'field' => 'active',
+                        'value' => '1',
+                    ]
                 ],
             ],
+            [
+                'id' => $inActiveStreamId,
+                'name' => 'test-inheritance',
+                'filters' => [
+                    [
+                        'type' => 'equals',
+                        'field' => 'active',
+                        'value' => '0',
+                    ]
+                ],
+            ]
+        ], Context::createDefaultContext());
+
+        $productStreamIndexer = $this->getContainer()->get(ProductStreamIndexer::class);
+        $productStreamIndexer->handle(
+            $productStreamIndexer->update($writtenEvent)
+        );
+
+
+        $productId = Uuid::randomHex();
+        $products = [$this->getProductData($productId)];
+
+        // Get product data [variantId => active]
+        $variantIds = [
+            Uuid::randomHex() => null,
+            Uuid::randomHex() => false,
+            Uuid::randomHex() => true,
+        ];
+
+        foreach ($variantIds as $id => $active) {
+            $productData = $this->getProductData($id);
+            $productData["parentId"] = $productId;
```

Comment:
we use only single quoted strings, you can also run `composer ecs-fix` to fix all code style issues automatically